packages/tools/viewer/src/viewer.ts

```diff
@@ -1124,11 +1149,14 @@ export class Viewer implements IDisposable {
         await this._loadModelLock.lockAsync(async () => {
             throwIfAborted(abortSignal, abortController.signal);
             this._activeModel?.dispose();
-            this._setActiveModel(null);
+            this._activeModelBacking = null;
             this.selectedAnimation = -1;
 
             if (source) {
-                this._setActiveModel(await this._loadModel(source, options, abortController.signal), Object.assign({ source, interpolateCamera: false }, options));
+                const model = await this._loadModel(source, options, abortController.signal);
+                model.makeActive(Object.assign({ source, interpolateCamera: false }, options));
+            } else {

```

Comment:
Sorry I added this `else` block but somehow I didn't notice that line 1152 already sets the _activeModelBacking to null, so I don't think this else block actually does anything.